salt/kube-apiserver/init.sls

```diff
@@ -2,6 +2,7 @@ include:
   - repositories
   - ca-cert
   - cert
+  - haproxy
```

Comment:
If we rely on the `top.sls` order as before, maybe this can be removed as well as the include in `haproxy/init.sls`, no?

Reply:
`top.sls` ordering still comes into effect, but `top.sls` ordering is just a hint to salt - it doesn't enforce anything. The only way to be sure state A has ran on a machine before state B is to encode that into an orchestration (which spreads the dependency chain out over the whole repo), or encode it into the states directly allowing any state to be called, and the right dependency ordering is generated by salt.

In this case, the `kube-apiserver-up` at the bottom of this file will fail if `haproxy` hasn't been setup - so we declare the dependency.

Reply:
However, the `include` alone also doesn't enforce any order (https://docs.saltstack.com/en/latest/ref/states/include.html, see the NOTE), we should also add a `require`.

Reply:
Ugh - salt's ordering, combined with how our states/highstates/orchestrations are laid out makes this *really* hard. heh